Tribler/Core/Modules/channel/channel_manager.py

```diff
@@ -0,0 +1,124 @@
+from binascii import hexlify
+import logging
+
+from Tribler.dispersy.taskmanager import TaskManager
+from Tribler.dispersy.util import blocking_call_on_reactor_thread, call_on_reactor_thread
+from Tribler.community.channel.community import ChannelCommunity
+
+from Tribler.Core.Modules.channel.channel import ChannelObject
+
+
+class ChannelManager(TaskManager):
+    """
+    The Manager class that handles the Channels owned by ourselves.
+    It supports multiple-Channel creation and RSS feed.
+    """
+
+    def __init__(self, session):
+        super(ChannelManager, self).__init__()
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self.session = session
+        self.dispersy = None
+
+        self._channel_mode_map = {u'open': ChannelCommunity.CHANNEL_OPEN,
+                                  u'semi-open': ChannelCommunity.CHANNEL_SEMI_OPEN,
+                                  u'closed': ChannelCommunity.CHANNEL_CLOSED,
+                                  }
+
+        self._channel_list = []
+
+    @blocking_call_on_reactor_thread
+    def initialize(self):
+        self.dispersy = self.session.get_dispersy_instance()
+
+        # get all channels owned by me
+        from Tribler.community.channel.community import ChannelCommunity
+        for community in self.session.lm.dispersy.get_communities():
+            if isinstance(community, ChannelCommunity) and community.master_member and community.master_member.private_key:
+                channel_obj = ChannelObject(self.session, community, is_created=True)
+                channel_obj.initialize()
+                self._channel_list.append(channel_obj)
+
+                self._logger.debug(u"loaded channel '%s', %s", channel_obj.name, hexlify(community.cid))
+
+    @blocking_call_on_reactor_thread
```

Comment:
@NielsZeilemaker There is no other threads involved here except the reactor thread, so it is safe to do this here.